File: application/controller/common_test.go
// Copyright (C) 2019-2026 Ni Rui <ranqus@gmail.com>
// Copyright (C) 2026 Snuffy2
// SPDX-License-Identifier: AGPL-3.0-only

package controller

import (
	"html"
	"testing"
)

func TestParseServerMessage(t *testing.T) {
	for _, test := range [][]string{
		{
			"<b>This is a [测试](http://example.com) " +
				"[for link support](http://example.com)</b>.",
			"&lt;b&gt;This is a " +
				"<a href=\"http://example.com\" target=\"_blank\">测试</a> " +
				"<a href=\"http://example.com\" target=\"_blank\">for link support</a>" +
				"&lt;/b&gt;.",
		},
		{
			"[测试](http://example.com)",
			"<a href=\"http://example.com\" target=\"_blank\">测试</a>",
		},
		{
			"[测试](http://example.com).",
			"<a href=\"http://example.com\" target=\"_blank\">测试</a>.",
		},
		{
			".[测试](http://example.com)",
			".<a href=\"http://example.com\" target=\"_blank\">测试</a>",
		},
	} {
		result := parseServerMessage(html.EscapeString(test[0]))
		if result != test[1] {
			t.Errorf("Expecting %v, got %v instead", test[1], result)
			return
		}
	}
}

File: ui/stream/common_test.js
// Copyright (C) 2019-2026 Ni Rui <ranqus@gmail.com>
// Copyright (C) 2026 Snuffy2
// SPDX-License-Identifier: AGPL-3.0-only

import assert from "assert";
import * as common from "./common.js";

describe("Common", () => {
  it("separateBuffer", async () => {
    let resultArr = [];
    const expected = new Uint8Array([
        0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3,
        4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7,
        8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1,
        2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 0, 1, 2, 3, 4, 5,
        6, 7, 8, 9, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9,
      ]),
      sepSeg = common.separateBuffer(expected, 16);

    sepSeg.forEach((d) => {
      resultArr.push(...d);
    });

    const result = new Uint8Array(resultArr);

    assert.deepStrictEqual(result, expected);
  });
});
